hazelcast-client-new/src/main/java/com/hazelcast/client/cache/impl/AbstractClientInternalCacheProxy.java

```diff
@@ -409,11 +556,35 @@ protected ClientInvocationFuture invoke(ClientMessage req, Data keyData, int com
         } catch (Exception e) {
             throw ExceptionUtil.rethrow(e);
         }
-        return new ClientDelegatingFuture<Boolean>(future, clientContext.getSerializationService(),
-                putIfAbsentResponseDecoder);
+        ClientDelegatingFuture delegatingFuture =
+                new ClientDelegatingFuture<Boolean>(future, clientContext.getSerializationService(),
+                                                    putIfAbsentResponseDecoder);
+        if (async && statisticsEnabled) {
+            delegatingFuture.andThen(new ExecutionCallback<Object>() {
```

Comment:
Can't we add this callback after `putIfAbsentAsyncInternal ` returned ?  They are also returning completable future. 
Async methods with isAsync boolean looks odd. I am basically trying to get rid of that boolean .
And my comment applies to all other async methods with async boolean.